### PR TITLE
Fix custom launch templates being ignored

### LINKED ISSUE #1193: eks.ManagedNodeGroup: launchTemplate is ignored
### What happened?

It looks like version 2.5.0 introduced a regression that results in the `launchTemplate` arg to `eks.ManagedNodeGroup` being ignored. The example below used to work fine up until and including version 2.4.0, but breaks from version 2.5.0 onwards. The launchTemplate arg is ignored and thus removed from the resource as shown in the output.

I believe the regression is caused by a change in https://github.com/pulumi/pulumi-eks/pull/1131; I'll add a comment to that PR.

### Example
```ts
new eks.ManagedNodeGroup("nodegroup-static", {
  cluster: cluster,
  amiType: "AL2_ARM_64",
  instanceTypes: ["m7g.xlarge"],
  subnetIds: vpc.privateSubnetIds,
  nodeRole: cluster.instanceRoles[0],
  launchTemplate: {  // <-- this is ignored
    id: launchTemplate.id,
    version: pulumi.interpolate`${launchTemplate.latestVersion}`,
  },
});

```

### Output of `pulumi about`
```
CLI          
Version      3.119.0
Go Version   go1.22.4
Go Compiler  gc

Plugins
KIND      NAME        VERSION
resource  aws         6.39.1
resource  awsx        2.11.0
resource  docker      4.5.4
resource  docker      3.6.1
resource  eks         2.5.0
resource  kubernetes  4.13.1
language  nodejs      unknown

Dependencies:
NAME                VERSION
@pulumi/awsx        2.11.0
@pulumi/eks         2.5.0
@pulumi/kubernetes  4.13.1
@pulumi/pulumi      3.119.0
@types/node         20.14.2
ts-node             10.9.2
yaml                2.4.5
@pulumi/aws         6.39.1
```

### Additional context
```
pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:prod::base::pulumi:pulumi:Stack::base-prod]
        ~ aws:eks/nodeGroup:NodeGroup: (update)
            [id=xxx-prod:nodegroup-static-prod20240304172943231900000001]
            [urn=urn:pulumi:prod::base::eks:index:ManagedNodeGroup$aws:eks/nodeGroup:NodeGroup::nodegroup-static]
            [provider=urn:pulumi:prod::base::pulumi:providers:aws::default_6_39_1::6de6c1e0-804e-479e-b4d1-e06c3f24e08b]
          - launchTemplate: {
              - id     : "lt-redacted"
              - name   : "nodegroup-static-prod20240228152929421500000001"
              - version: "5"
            }
          - launchTemplate: {
              - id     : "lt-redacted"
              - name   : "nodegroup-static-prod20240228152929421500000001"
              - version: "5"
            }
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by flostadler**
### Proposed changes

In version 2.5.0 custom launch templates got broken. This re-enables those and adds a regression test.

### Related issues (optional)

fixes #1193

### CODE REVIEW

File: examples/examples_nodejs_test.go [hunk 1]
```diff
@@ -760,3 +760,34 @@ func TestAccAuthenticationModeMigration(t *testing.T) {
 
 	integration.ProgramTest(t, &test)
 }
+
+func TestAccManagedNodeGroupCustom(t *testing.T) {
```
Comment (rquitales) [inline examples/examples_nodejs_test.go:764]: Can you add this to our CI configs as well?

**Review by rquitales — APPROVED** [review]
Thanks for the fix!

File: examples/examples_nodejs_test.go (hunk 1 above)
Comment (flostadler) [reply examples/examples_nodejs_test.go:764]: It's included in this one: https://github.com/pulumi/pulumi-eks/actions/runs/9501673502/job/26188058924?pr=1196#step:23:1419

Comment (rquitales) [reply examples/examples_nodejs_test.go:764]: Ah yes, the name matching is not a exact match:
https://github.com/pulumi/pulumi-eks/blob/92465dce68dd560ae48fd0211cae9a74b4ee19e4/Makefile#L196C3-L197C178

**Pull request merged**